### PR TITLE
Migrate to non-blocking architecture using Spring WebFlux

### PR DESCRIPTION
**Pull request by SlimaneAmar**
Migrate to non-blocking architecture using Spring WebFlux

### CODE REVIEW

File: src/main/java/org/gridsuite/modification/server/NetworkModificationService.java [hunk 1]
```diff
@@ -8,75 +8,85 @@
...
 
             networkStoreService.flush(network);
 
-            return Set.of(sw.getVoltageLevel().getSubstation().getId());
-        } else {
-            return Set.of();
+            return listener.getModifications();
+        } catch (Exception e) {
+            NetworkModificationException exc = new NetworkModificationException(typeIfError, e.getMessage());
+            LOGGER.error(exc.getMessage());
```
Comment (jonenst) [inline src/main/java/org/gridsuite/modification/server/NetworkModificationService.java:55]: Here we lose the original stacktrace. Just add e as the cause of exc ?

File: src/main/java/org/gridsuite/modification/server/NetworkModificationService.java [hunk 2]
```diff
@@ -8,75 +8,85 @@
...
-        GroovyShell shell = new GroovyShell(binding, conf);
+    private Mono<Network> getNetwork(UUID networkUuid) {
         try {
-            shell.evaluate(groovyScript);
-            networkStoreService.flush(network);
-
-            return new GroovyScriptResult(true, listener.getModifications());
-        } catch (Exception ignored) {
-            return new GroovyScriptResult(false, listener.getModifications());
+            return Mono.just(networkStoreService.getNetwork(networkUuid));
```
Comment (jonenst) [inline src/main/java/org/gridsuite/modification/server/NetworkModificationService.java:62]: use a thread pool to perform this blocking work ?

File: src/main/java/org/gridsuite/modification/server/NetworkModificationService.java [hunk 3]
```diff
@@ -8,75 +8,85 @@
...
+                .map(network -> doModification(network, () -> network.getSwitch(switchId).setOpen(open)))
+                .switchIfEmpty(Mono.just(Set.of()));
+    }
+
+    private Mono<Void> assertGroovyScriptNotEmpty(String groovyScript) {
+        return StringUtils.isBlank(groovyScript) ? Mono.error(new NetworkModificationException(GROOVY_SCRIPT_EMPTY)) : Mono.empty();
+    }
+
+    Mono<Set<String>> applyGroovyScript(UUID networkUuid, String groovyScript) {
+        return Mono.when(assertGroovyScriptNotEmpty(groovyScript)).then(
```
Comment (jonenst) [inline src/main/java/org/gridsuite/modification/server/NetworkModificationService.java:82]: why Mono.when ?

Comment (SlimaneAmar) [reply src/main/java/org/gridsuite/modification/server/NetworkModificationService.java:82]: Removed
At the beginning there were several assertions

File: src/main/java/org/gridsuite/modification/server/NetworkModificationService.java (hunk 1 above)
Comment (SlimaneAmar) [reply src/main/java/org/gridsuite/modification/server/NetworkModificationService.java:55]: Done

File: src/main/java/org/gridsuite/modification/server/NetworkModificationService.java (hunk 2 above)
Comment (SlimaneAmar) [reply src/main/java/org/gridsuite/modification/server/NetworkModificationService.java]: Done

File: src/main/java/org/gridsuite/modification/server/NetworkModificationService.java (hunk 1 above)
Comment (SlimaneAmar) [reply src/main/java/org/gridsuite/modification/server/NetworkModificationService.java:56]: Done

**Review by FranckLecuyer — APPROVED** [review]

**Review by jonenst — APPROVED** [review]

**Pull request merged**